drivers/mpsl/flash_sync/flash_sync_mpsl.c

```diff
@@ -80,9 +81,9 @@ timeslot_callback(mpsl_timeslot_session_id_t session_id, uint32_t signal)
 
 	switch (signal) {
 	case MPSL_TIMESLOT_SIGNAL_START:
-		if (_context.op_desc->handler(_context.op_desc->context) ==
-		    FLASH_OP_DONE) {
-			_context.status = 0;
+		rc = _context.op_desc->handler(_context.op_desc->context);
+		if (rc != FLASH_OP_ONGOING) {
+			_context.status = (rc == FLASH_OP_DONE) ? 0 : rc;
```

Comment:
`_context.status` should be something from `errno.h` negated. `FLASH_OP_ABORTED` will not do.

Reply:
fixed in the handler() patch